### PR TITLE
Spinner: refactor to move from inheritance to composition

### PR DESCRIPTION
**Pull request by filippobrizzi**
# Description
Spinner class now follows the composition pattern. 

Update MessageQueueConsumer to use the Spinner class instead of re-implementing the logic.

### CODE REVIEW

File: modules/concurrency/examples/spinner_example.cpp [hunk 1]
```diff
@@ -2,37 +2,50 @@
...
 #include "hephaestus/concurrency/spinner.h"
 #include "hephaestus/utils/signal_handler.h"
 
-class TestSpinner : public heph::concurrency::Spinner {
+static constexpr auto RATE_HZ = 10;
+
+class Worker {
 public:
-  std::atomic<int> counter = 0;
+  Worker() : spinner_([this] { doWork(); }, RATE_HZ) {
```
Comment (johannes-graeter) [inline modules/concurrency/examples/spinner_example.cpp:18]: great design!

File: modules/concurrency/include/hephaestus/concurrency/message_queue_consumer.h [hunk 2]
```diff
@@ -18,34 +18,44 @@ class MessageQueueConsumer {
 public:
   using Callback = std::function<void(const T&)>;
   [[nodiscard]] MessageQueueConsumer(Callback&& callback, std::optional<std::size_t> max_queue_size);
-  ~MessageQueueConsumer();
+  ~MessageQueueConsumer() = default;
```
Comment (johannes-graeter) [inline modules/concurrency/include/hephaestus/concurrency/message_queue_consumer.h:21]: nice !

File: modules/concurrency/tests/spinner_tests.cpp [hunk 3]
```diff
@@ -50,34 +37,37 @@ TEST(SpinnerTest, SpinTest) {
...
 
-TEST(SpinnerTest, StopCallbackTest) {
-  TestSpinner spinner;
-  std::atomic<bool> callback_called(false);
+TEST(SpinnerTest, StopCallback) {
+  static constexpr auto WAIT_FOR = std::chrono::milliseconds{ 10 };
 
-  spinner.addStopCallback([&]() { callback_called.store(true); });
+  std::size_t callback_called_counter = 0;
+  Spinner spinner([&callback_called_counter] { ++callback_called_counter; });
```
Comment (johannes-graeter) [inline modules/concurrency/tests/spinner_tests.cpp:47]: nit: interesting didn't know you could omit the () ... maybe still add them? for me this just looks strange ;)

**Review by johannes-graeter — APPROVED** [review]
Great! Love the new design!

**Comment by lhruby** [comment]
Nice!

**Review by floriantschopp — APPROVED** [review]

**Pull request merged**